### PR TITLE
Embed API port to etcd

### PR DESCRIPTION
**Pull request by huachaohuang**
Remove http-addr (9090).

### CODE REVIEW

File: server/api/server.go [hunk 1]
```diff
@@ -16,28 +16,23 @@ package api
...
 
-	"github.com/juju/errors"
 	"github.com/pingcap/pd/server"
 	"github.com/urfave/negroni"
 )
 
-// ServeHTTP creates a HTTP service and serves.
-func ServeHTTP(addr string, svr *server.Server) error {
+// NewHandler creates a HTTP handler for API.
+func NewHandler(prefix string, svr *server.Server) http.Handler {
```
Comment (siddontang) [inline server/api/server.go]: no need prefix, use const "/pd" directly.

**Comment by siddontang** [comment]
@iamxy we should update all documents, docker later.

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @qiuyesuifeng

**Comment by siddontang** [comment]
PTAL @shenli @overvenus

**Comment by overvenus** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
@huachaohuang  We can remove pd's  /metrics router, for prometheus is a global variable, we can directly use etcd's  /metrics for it.

**Pull request merged**